scalyr_agent/builtin_monitors/url_monitor.py

```diff
@@ -103,11 +107,62 @@ def _initialize(self):
         else:
             self.extractor = None
 
-    def gather_sample(self):
+    @staticmethod
```

Comment:
Why have this as a static method?  You wouldn't have to pass in things like `logger` or `url`, etc if it was a normal method.  I take it is for testing?  How about we just require the tests to create an instead of `UrlMonitor`?

Reply:
yes, you're right - it was mainly for more testability. However, we can instantiate an object in the test, no big deal.